compiler/rustc_errors/src/styled_buffer.rs

```diff
@@ -49,29 +66,31 @@ impl StyledBuffer {
     }
 
     fn ensure_lines(&mut self, line: usize) {
-        while line >= self.text.len() {
-            self.text.push(vec![]);
-            self.styles.push(vec![]);
+        while line >= self.lines.len() {
+            self.lines.push(vec![]);
         }
     }
 
+    /// Sets `chr` with `style` for given `line`, `col`.
+    /// If line not exist in `StyledBuffer`, adds lines up to given
+    /// and fills last line with spaces and `Style::NoStyle` style
     pub fn putc(&mut self, line: usize, col: usize, chr: char, style: Style) {
         self.ensure_lines(line);
-        if col < self.text[line].len() {
-            self.text[line][col] = chr;
-            self.styles[line][col] = style;
+        if col < self.lines[line].len() {
```

Comment:
can we replace that large if statement with
```suggestion
        if col >= self.lines[line].len() {
            self.lines[line].resize(col + 1, StyledChar::new(' ', Style::NoStyle);
        }
        self.lines[line][col] = StyledChar::new(chr, style);
```

Reply:
Fixed.